### ISSUE TITLE
Login page broken in 3.2

### ISSUE THREAD

**Issue by brendanheywood** [catalyst/moodle-auth_enrolkey#15]
HI Brendan, 

Great pluggin! 
Tested it for 3.2 but after visiting: /login/signup.php? i get the error
Can not render widget, renderer method (render_enrolkey_signup_form) not found. 
I know it is not supported yet for 3.2, any idea on when it will be?

Regards Rob

**Comment by troyfennell** [comment]
Hi Brendan and Rob,

I am having an issue with the login page too. I am not sure what issue you are referring when you say ""the error". However, my students, who have been enrolled with the enrol plugin cannot reset their passwords. I know this is due to the plugin, because when i alter their authentication method to another option such as manual accounts, the reset functionality restores.

This plugin is a cornerstone in our LMS. Please, help me to fix this issue.

**Comment by yjeanrenaud** [comment]
any news?

**Comment by brendanheywood** [comment]
Just echoing the private email: if you'd like to sponsor this then that would be great :)

**Comment by dheuring** [comment]
What will it take to get this working for Moodle 3.2?

**Comment by brendanheywood** [comment]
Not sure exactly what's busted in 3.2 but it's probably a couple hours

**Comment by mwithheld** [comment]
Stack trace (plugin version=2016091900; Moodle version=3.3.1+ (Build: 20170907))
line 212 of /lib/outputrenderers.php: coding_exception thrown
line 108 of /login/signup.php: call to renderer_base->render()

Disabling the new Moodle core signup form renderer makes it work again:
login/signup.php::~line 100 - I added the 'false && ' at the beginning:
```
if (false && $mform_signup instanceof renderable) {
    // Try and use the renderer from the auth plugin if it exists.
    try {
        $renderer = $PAGE->get_renderer('auth_' . $authplugin->authtype);
    } catch (coding_exception $ce) {
        // Fall back on the general renderer.
        $renderer = $OUTPUT;
    }
    echo $renderer->render($mform_signup);
} else {
    // Fall back for auth plugins not using renderables.
    $mform_signup->display();
}
```

**Comment by mwithheld** [comment]
The only other hitch I see w/ Moodle 3.3 is the captcha shows below the submit/cancel buttons (using the Boost theme w/ no added CSS/JS)

**Comment by dheuring** [comment]
This solution appears to work for the students.  It creates their accounts and enrolls them in the course quite nicely.  I'm using Moodle 3.2.1.  However, when I went to install an update to a completely different plugin, I got error messages and was essentially stuck until I could remove enrollment key as the default self-registration setting on the Authentication page.  Evidently, those lines in the signup.php that are disabled are needed when doing a plugin upgrade, at least for me.  In all fairness, hacking a core program file is not a best practice.  It can't be that complicated to get this plugin working again.  I contacted the developers about it but it was too pricey for my organization so we're just continuing to use the longer process to get students in our courses.
